### ISSUE TITLE
Generated output with `staticHelpers: true` seems invalid

### ISSUE THREAD

**Issue by mydea** [embroider-build/embroider#714]
Forked this out from https://github.com/stefanpenner/ember-strict-resolver/pull/45

When building an app with strict helpers, I noticed that e.g. `require('helper:page-title')` would not return the normal module with a default property, but directly return the default export itself (!?). Not sure what exactly is going on, but after some digging I noticed this small difference in output:

## With `staticHelpers: true`

```js
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */
__webpack_require__.d(__webpack_exports__, 'default ', function () {
  return PageTitle;
});
```

## With `staticHelpers: false`

```js
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */
__webpack_require__.d(__webpack_exports__, 'default', function () {
  return PageTitle;
});
```

Notice the space after default in the first example.

**Comment by rwjblue** [comment]
That `'default '` string seems wrong to me, but I'm not sure exactly internally where we handle emitting those modules.

**Comment by ef4** [comment]
That certainly looks weird, but I will point out that using `require` to access helpers when `staticHelpers` is on is highly likely to break.

Right now if you try it after a template has evaluated that uses the given helper, it will work. If you try it before any template that uses the helper evalutes, it will not.

In upcoming releases we will take advantage of ember's template lexical scope and then it will never work -- the helper will never be registered with the runtime AMD loader.

**Comment by rwjblue** [comment]
@ef4 - This is the resolver resolving the helper, is there some override for `ember-resolver` in these cases?

**Comment by ef4** [comment]
No, it does go through the resolver internally. I just mean people shouldn't rely on looking in requirejs.entries themselves.

**Comment by rwjblue** [comment]
> I just mean people shouldn't rely on looking in requirejs.entries themselves.

Yes, totally agreed!

**Comment by mydea** [comment]
Yes, to clarify, that has nothing to do with manually using `require()`, it has the same output with just a regular app using staticHelpers, even with the default ember-resolver. Only that ember-resolver seems to handle this by also allowing the module returned from `require()` to be used directly (instead of `module['default']`), which makes this work, although I am still not sure why that would even work that way...
